Core/include/Acts/Seeding2/TripletSeedFinder.hpp

```diff
@@ -0,0 +1,229 @@
+
+    SpacePointIndex2 spacePoint() const {
+      return m_container->m_topSpacePoints[m_index];
+    }
+
+    float curvature() const { return m_container->m_curvatures[m_index]; }
+
+    float impactParameter() const {
+      return m_container->m_impactParameters[m_index];
+    }
+
+   private:
+    const TripletTopCandidates* m_container{};
+    Index m_index{};
+  };
+
+  Proxy operator[](std::size_t index) const { return Proxy(this, index); }
+
+  using const_iterator =
+      ContainerIterator<TripletTopCandidates, Proxy, Index, true>;
+
+  const_iterator begin() const { return const_iterator(*this, 0); }
+  const_iterator end() const { return const_iterator(*this, size()); }
+
+ private:
+  std::vector<SpacePointIndex2> m_topSpacePoints;
+  std::vector<float> m_curvatures;
+  std::vector<float> m_impactParameters;
+};
+
+class TripletSeedFinder {
+ public:
+  struct Config {
+    /// Minimum transverse momentum (pT) used to check the r-z slope
+    /// compatibility of triplets with maximum multiple scattering effect
+    /// (produced by the minimum allowed pT particle) + a certain uncertainty
+    /// term. Check the documentation for more information
+    /// https://acts.readthedocs.io/en/latest/core/reconstruction/pattern_recognition/seeding.html
+    float minPt = 400 * UnitConstants::MeV;
+    /// Number of sigmas of scattering angle to be considered in the minimum pT
+    /// scattering term
+    float sigmaScattering = 5;
+    /// Term that accounts for the thickness of scattering medium in radiation
+    /// lengths in the Lynch & Dahl correction to the Highland equation default
+    /// is 5%
+    float radLengthPerSeed = 0.05;
+    /// Maximum transverse momentum for scattering calculation
+    float maxPtScattering = 10 * UnitConstants::GeV;
+    /// Maximum value of impact parameter estimation of the seed candidates
+    float impactMax = 20 * UnitConstants::mm;
+    /// Parameter which can loosen the tolerance of the track seed to form a
+    /// helix. This is useful for e.g. misaligned seeding.
+    float helixCutTolerance = 1;
+
+    /// Tolerance parameter used to check the compatibility of space-point
+    /// coordinates in xyz. This is only used in a detector specific check for
+    /// strip modules
+    float toleranceParam = 1.1 * UnitConstants::mm;
+
+    /// Delegates for accessors to detailed information on double measurement
+    /// that produced the space point. This is mainly referring to space points
+    /// produced when combining measurement from strips on back-to-back modules.
+    /// Enables setting of the following delegates.
+    bool useStripInfo = false;
+
+    /// Whether the input doublets are sorted by cotTheta
+    bool sortedByCotTheta = true;
+  };
+
+  struct DerivedConfig : public Config {
+    DerivedConfig(const Config& config, float bFieldInZ);
+
+    float bFieldInZ = std::numeric_limits<float>::quiet_NaN();
+    float highland = std::numeric_limits<float>::quiet_NaN();
```

Comment:
Default changes from 0 to `nan`, but I imagine these are set from the constructor anyway?

Reply:
right these are set on construction. I just thought `0` makes little sense if it ever ends up as a value. I would rather let everything fail in this case